src/api/messages/getMessages.js

```diff
@@ -1,20 +1,29 @@
 /* @flow */
-import type { Auth, Message, Narrow } from '../../types';
+import type { Auth, ApiResponseSuccess, Message, Narrow } from '../apiTypes';
 import { apiGet } from '../apiFetch';
 
+type ApiResponseMessages = ApiResponseSuccess & {
+  anchor: number,
+  found_anchor: boolean,
+  found_newest: boolean,
+  found_oldest: boolean,
+  messages: Message[],
+};
+
 export default async (
   auth: Auth,
   narrow: Narrow,
   anchor: number,
   numBefore: number,
   numAfter: number,
   useFirstUnread: boolean = false,
-): Promise<Message[]> =>
-  apiGet(auth, 'messages', res => res.messages, {
+  applyMarkdown: boolean = true,
+): Promise<ApiResponseMessages> =>
+  apiGet(auth, 'messages', (res: ApiResponseMessages): ApiResponseMessages => res, {
```

Comment:
It'd be nice to get to leave out all the repetition in this `resFunc` argument.

What if you write just `res => res` -- or perhaps `undefined`, which gets you the default which is the same thing? We have a type signature on the outer function already, so its own type should come out just as clearly, right?

Reply:
I was thinking something similar. I will remove each such function in one function per commit.

Reply:
#2933